docs/api/apiv3/components/schemas/watchers_model.yml

```diff
@@ -1,6 +1,43 @@
 # Schema: WatchersModel
 ---
 type: object
+properties:
```

Comment:
Technically seems to work, but causes the `WorkPackagesModelLinks` class to be called `WorkPackagesModelAllOfLinks`, which is not really appropriate / meaningful.

Also causes generation of addition classes `WorkPackagesModelAllOfEmbedded` which haven't been there before.

I'll have to take some more time to check if this makes sense, and maybe how to influence name generation.

In general I agree that your approach reduces redundancy in the schema definitions.

Reply:
Can you perchance provide me a glance into those generated classes? I admit, I'm kind of confused, why Java should do it this way, as it could simply create a class `WorkPackagesModel`, that has all the merged properties of `Link` and the ones mentioned there. No idea, why taking the `allOf` into the name. Or why creating an `embedded`, at all.

Reply:
Concerning the `WorkPackagesModel` it does exactly what you expected - this class is generated and contains the Collection Model attributes plus `_link` and `_embedded`.

I also think I understand now what happens here, but didn't have time so far to think about an alternative.

For embedded object definitions like `_link` or `_embedded` in the above example, additional classes representing these types are generated. These classes get the name `<ContainerClassName><CleanedAttributeName>` - so e.g. (just examples, these classes do NOT exists here in this case, see below) `WorkPackagesModelLinks` or `WorkPackagesModelEmbedded`. So far, so logical.

(Side node: My last analysis was on a wrong track concerning the generated "*Embedded" class - I should have taken a second look before replying. The class is justified, represents the object type defined for the `_embedded` attribute, and was only newly generated after your changes suggested above because before, the type defined for the `_embedded` attribute perfectly matched the type structure for the type of the "Available Assignees Model"'s `_embedded` attribute - and the generator recognized that and re-used the latter inside the `WorkPackagesModel`. After I applied your suggested changes to "Work Packages Model", this was no longer the case, and the additional `*Embedded` type for the Work Package's `_embedded` attribute had to be generated.)

Where it get's somewhat strange now is the (anonymous) object definition directly nested into the `allOf:` in your suggestion.

This causes the OpenAPI CodeGen to **also** (besides the expected `WorkPackagesModel` class) generate an `WorkPackagesModelAllOf` class, which only contains a `_links` and an `_embedded` attribute.

This class is then actually used / referenced from nowhere. What seems to happen here is that the same logic as for object types that are defined for attributes (like `_embedded` in the above example) is used, and instead of a property name, which is not available in this case, the `allOf` is used to generate the type suffix.

Compare the following snippets:

```yml
_links:
  type: object
```

This causes a class `<ContainerTypeName>Links` to be generated for the object definition specified in the `_links` attribute.

```yml
allOf:
  - "$ref": "./collection_model.yml"
  - type: object
```

This then (strangely?) causes a class `<ContainerTypeName>AllOf` to be generated - looks analogous to the name generation in the case before, except in the case of an attribute the type definition is nested in, this makes sense, while less so if it's nested in an `allOf` context...

However I still don't have sufficient insight into OpenAPI V3 in order to determine if this makes sense somehow, or if it's a bug in OpenAPI CodeGen.

Then it gets logical again - `WorkPackagesModelAllOf` is the direct container type of the `_links` and `_embedded` attributes, so the types generated for these are consequentually names `WorkPackagesModelAllOfLinks` and `WorkPackagesModelAllOfEmbedded` - which are somewhat strange names, unfortunately.

These two types are then also used for the `links` and `embedded` attributes in the generated `WorkPackagesModel` (without `AllOf` suffix) class.

Besides that, the generated code is as expected, though...

Reply:
So, about question 2:

Here both examples are misleading and wrong currently (those in `users_model.yml` and those in `watchers_model.yml`). Your premise that the watchers endpoint actually returns a collection of users, is correct. I would like to update those examples in the collection models on top of your PR, so when it is merged, I will create another one to make some polishing to `user_model.yml`, which is mostly about some missing or wrongly typed properties. and amend the examples.

Reply:
Sorry for the double-post -
1. first I could not add the first of these two coments (tried three times),
2. then I could not edit it to remove the doublicate content which crept in during the submission retries (the content does not change after editing)
3. and now I cannot delete it (nothing happens if I select "delete" from its menu)...

Strange.

Reply:
Uhm, I guess there is a bit of duplication? :D

Yes, please do not merge both collection models. There IS a difference in the collection models, in the `_links` they show (on the collection, not the embedded user elements). The user elements are exactly the same in both, even use the same representer code. Hence, we can, like you have it already, `$ref` the `user_model.yml` in the `watchers_model.yml`.

Reply:
> edith says: At least we have some funny history in here now ;)

I feel innocent... ;)